### PR TITLE
feat(arcgis): auth UI

### PR DESCRIPTION
**Pull request by mspivak-actionengine**

### CODE REVIEW

File: src/components/action-icon-button/action-icon-button.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  border-radius: 4px;
+  align-items: center;
+  justify-content: center;
+  margin-right: 16px;
+  `;
+
+export const ActionIconButton = (props: ActionIconButtonProps) => {
+
+  const grayed = props.style === "disabled";
+  const StyledIcon = styled(props.icon) <{ grayed?: boolean }>`
```
Comment (belom88) [inline src/components/action-icon-button/action-icon-button.tsx:73]: Use `useMemo` here

**Review by belom88 — CHANGES_REQUESTED** [review]
I'd use for the commit message "feat(arcgis): auth UI"

File: src/app.tsx [hunk 2]
```diff
@@ -106,6 +119,18 @@ const THEMES: AppThemes = {
       validateTileOk: color_brand_secondary,
       validateTileWarning: color_accent_primary,
       filtrationImage: color_canvas_secondary,
+
+      actionIconButtonDisabledColor: color_canvas_secondary,
+      actionIconButtonDisabledBG: `${color_brand_tertiary}66`,
+      actionIconButtonDisabledBGHover: `${color_brand_tertiary}33`,// TODO: waiting for the design...
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/app.tsx:125]: There's no "hover" for "disabled button" (both dark and light)

File: src/components/logout-button/logout-button.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  }
+`;
+
+export const LogoutButton = ({
+  children,
+  onClick,
+}: LogoutButtonProps) => {
+  return (
+    <Button onClick={onClick}>
+      <ButtonText>{children}</ButtonText>
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:44]: Done

File: src/components/logout-button/logout-button.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+type LogoutButtonProps = {
+  children?: React.ReactNode;
+  onClick?: () => void;
+};
+
+const LogoutImage = styled(LogoutIcon)`
+  display: flex;
+  position: relative;
+  height: 14px;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:12]: Removed "height". Relying on the svg properties.

File: src/components/logout-button/logout-button.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import styled from "styled-components";
+import LogoutIcon from "../../../public/icons/logout.svg";
+
+type LogoutButtonProps = {
+  children?: React.ReactNode;
+  onClick?: () => void;
+};
+
+const LogoutImage = styled(LogoutIcon)`
+  display: flex;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:10]: Removed.

File: src/components/logout-button/logout-button.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    }
+  }
+`;
+
+export const LogoutButton = ({
+  children,
+  onClick,
+}: LogoutButtonProps) => {
+  return (
+    <Button onClick={onClick}>
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:43]: Done.

File: src/components/action-icon-button/action-icon-button.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  icon: StyledComponent<any, DefaultTheme, object, string | number | symbol>;
+  buttonSize: ButtonSize;
+  style?: "active" | "disabled";
+  onClick?: () => void;
+};
+
+/*
+  https://developer.mozilla.org/en-US/docs/Web/CSS/opacity
+
+  opacity applies to the element as a whole, including its contents, even though the value is not inherited by child elements.
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:17]: Removed.

File: src/components/action-icon-button/action-icon-button.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import styled from "styled-components";
+import { ButtonSize } from "../../types";
+import { StyledComponent, DefaultTheme } from "styled-components";
+
+type ActionIconButtonProps = {
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:6]: Done.

File: src/components/action-icon-button/action-icon-button.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+  children?: React.ReactNode;
+  icon: StyledComponent<any, DefaultTheme, object, string | number | symbol>;
+  buttonSize: ButtonSize;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:9]: Done.

File: src/components/action-icon-button/action-icon-button.tsx [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+      : theme.colors.actionIconButtonActiveBG
+  )};
+
+  border-radius: 4px;
+  align-items: center;
+  justify-content: center;
+  margin-right: 16px;
+  `;
+
+export const ActionIconButton = (props: ActionIconButtonProps) => {
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:70]: Done.

File: src/components/action-icon-button/action-icon-button.tsx [hunk 11]
```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import styled from "styled-components";
+import { ButtonSize } from "../../types";
+import { StyledComponent, DefaultTheme } from "styled-components";
+
+type ActionIconButtonProps = {
+  children?: React.ReactNode;
+  icon: StyledComponent<any, DefaultTheme, object, string | number | symbol>;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:8]: Done.

File: src/components/action-icon-button/action-icon-button.tsx (hunk 1 above)
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:73]: Done.

File: src/app.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/app.tsx]: Done including hover

File: src/components/layers-panel/layers-control-panel.tsx [hunk 12]
```diff
@@ -24,6 +29,16 @@ type LayersControlPanelProps = {
   deleteLayer: (id: string) => void;
 };
 
+const EsriStyledImage = styled(EsriImage)`
+  display: flex;
+  position: relative;
+  top: 0.37px;
+  width: 41.9px;
+  height: 15.65px;
```
Comment (belom88) [inline src/components/layers-panel/layers-control-panel.tsx:37]: These properties seems not needed

File: src/components/layers-panel/layers-control-panel.tsx [hunk 13]
```diff
@@ -178,12 +208,36 @@ export const LayersControlPanel = ({
...
+        )}
+        {showImport && (
+          <ActionIconButtonContainer>
+            <ActionIconButton Icon={ImportIcon} style={'active'} size={ButtonSize.Small} onClick={onArcGisImportClick}>
+              Import from ArcGIS
+            </ActionIconButton>
+            <EsriStyledImage />
+          </ActionIconButtonContainer>
+        )}
+        {showLogout && (
```
Comment (belom88) [inline src/components/layers-panel/layers-control-panel.tsx:236]: Extra complexity is not needed
```suggestion
        {isLoggedIn&& (
```

File: src/components/layers-panel/layers-control-panel.tsx [hunk 14]
```diff
@@ -178,12 +208,36 @@ export const LayersControlPanel = ({
...
+          </ActionIconButtonContainer>
+        )}
+        {showImport && (
+          <ActionIconButtonContainer>
+            <ActionIconButton Icon={ImportIcon} style={'active'} size={ButtonSize.Small} onClick={onArcGisImportClick}>
+              Import from ArcGIS
+            </ActionIconButton>
+            <EsriStyledImage />
+          </ActionIconButtonContainer>
+        )}
```
Comment (belom88) [inline src/components/layers-panel/layers-control-panel.tsx:235]:
```suggestion
          <ActionIconButtonContainer>
            <ActionIconButton Icon={ImportIcon} style={isLoggedIn ? 'active' : 'disabled'} size={ButtonSize.Small} onClick={onArcGisActionClick}>
              {!isLoggedIn && "Login to ArcGIS"}
              {isLoggedIn && "Import from ArcGIS"}
            </ActionIconButton>
            <EsriStyledImage />
          </ActionIconButtonContainer>
```

File: src/components/action-icon-button/action-icon-button.tsx [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+  const grayed = useMemo(() => style === "disabled", [style]);
+
+  const StyledIcon = styled(Icon) <{ grayed?: boolean }>`
+    position: absolute;
+    fill: ${({ theme, grayed }) => (
+      grayed
+        ? theme.colors.actionIconButtonDisabledColor
+        : color_brand_tertiary
+    )};
+    `;
```
Comment (belom88) [inline src/components/action-icon-button/action-icon-button.tsx:82]:
```suggestion
  const StyledIcon = useMemo(() => styled(Icon) <{ grayed?: boolean }>`
    position: absolute;
    fill: ${({ theme, grayed }) => (
      grayed
        ? theme.colors.actionIconButtonDisabledColor
        : color_brand_tertiary
    )
    `}, [Icon]);
    
```

File: src/components/logout-button/logout-button.tsx [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+export const LogoutButton = ({
+  children,
+  onClick,
+}: LogoutButtonProps) => {
+  return (
+    <ButtonContainer>
+      <ButtonText>{children}</ButtonText>
+      <IconContainer onClick={onClick}>
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:56]: Done

File: src/components/logout-button/logout-button.tsx [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+  top: 1px;
+  stroke: ${color_canvas_secondary_inverted};
+`;
+
+type LogoutButtonProps = {
+  children?: React.ReactNode;
+  onClick?: () => void;
+};
+
+export const LogoutButton = ({
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:49]: Done

File: src/components/logout-button/logout-button.tsx [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+  padding-left: 10px;
+  transform: translate(0, -8px);
+
+  &:hover {
+    > :nth-child(2) > * {
+        stroke: ${({ theme }) => (theme.colors.logoutButtonIconColorHover)};
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:16]: Done.

File: src/components/logout-button/logout-button.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+  onClick?: () => void;
+};
+
+export const LogoutButton = ({
+  children,
+  onClick,
+}: LogoutButtonProps) => {
+  return (
+    <ButtonContainer>
+      <ButtonText>{children}</ButtonText>
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:55]: ButtonText is renamed to UserInfo.

File: src/components/logout-button/logout-button.tsx [hunk 20]
```diff
@@ -0,0 +1,61 @@
...
+  children?: React.ReactNode;
+  onClick?: () => void;
+};
+
+export const LogoutButton = ({
+  children,
+  onClick,
+}: LogoutButtonProps) => {
+  return (
+    <ButtonContainer>
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:54]: Done.

File: src/components/logout-button/logout-button.tsx [hunk 21]
```diff
@@ -0,0 +1,61 @@
...
+import {
+  color_canvas_secondary_inverted,
+} from "../../constants/colors";
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: flex-start;
+  align-items: center;
+  padding-left: 10px;
+  transform: translate(0, -8px);
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:12]: Done.

File: src/components/logout-button/logout-button.tsx [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+
+  &:hover {
+    > :nth-child(2) > * {
+        stroke: ${({ theme }) => (theme.colors.logoutButtonIconColorHover)};
+      }
+  }
+`;
+
+const ButtonText = styled.div`
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:21]: Done.

File: src/components/logout-button/logout-button.tsx [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+  position: relative;
+  height: 17px;
+  margin-left: 40px;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:24]: Done.

File: src/components/logout-button/logout-button.tsx [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+
+const IconContainer = styled.div`
+  position: relative;
+  width: 63px;
+  height: 17px;
+  cursor: pointer;
+`;
+
+const StyledIcon = styled(LogoutIcon)`
+  position: absolute;
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:39]: Done

File: src/app.tsx [hunk 25]
```diff
@@ -160,3 +182,4 @@ export const App = () => {
     </>
   );
 };
+
```
[the remark this answers is not in the document]
Comment (mspivak-actionengine) [reply src/app.tsx:185]: Done

File: src/components/layers-panel/layers-control-panel.tsx (hunk 14 above)
Comment (mspivak-actionengine) [reply src/components/layers-panel/layers-control-panel.tsx:235]: Done

File: src/components/logout-button/logout-button.tsx [hunk 26]
```diff
@@ -0,0 +1,61 @@
+import styled from "styled-components";
```
Comment (belom88) [inline src/components/logout-button/logout-button.tsx:1]: re-name the file to align with component's name

File: src/components/action-icon-button/action-icon-button.tsx [hunk 27]
```diff
@@ -0,0 +1,87 @@
+import React, { useMemo } from "react";
```
Comment (belom88) [inline src/components/action-icon-button/action-icon-button.tsx:1]: unused useMemo

File: src/components/action-icon-button/action-icon-button.tsx [hunk 28]
```diff
@@ -0,0 +1,87 @@
...
+          grayed
+            ? theme.colors.actionIconButtonDisabledColor
+            : color_brand_tertiary
+        } />
+      </IconContainer>
+      <ButtonText grayed={grayed}>{children}</ButtonText>
+    </Button>
+  );
+};
+
```
Comment (belom88) [inline src/components/action-icon-button/action-icon-button.tsx:87]: remove

**Review by belom88 — CHANGES_REQUESTED** [review]
Unnecessary scroll always seen:
image

File: src/components/logout-button/logout-button.tsx (hunk 26 above)
Comment (mspivak-actionengine) [reply src/components/logout-button/logout-button.tsx:1]: Done

File: src/components/action-icon-button/action-icon-button.tsx (hunk 27 above)
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:1]: Done

File: src/components/action-icon-button/action-icon-button.tsx (hunk 28 above)
Comment (mspivak-actionengine) [reply src/components/action-icon-button/action-icon-button.tsx:87]: Done

**Review by belom88 — APPROVED** [review]

**Pull request merged**